cf-oscore/src/main/java/org/eclipse/californium/oscore/OSCoreCtx.java

```diff
@@ -79,8 +80,47 @@
 
 	private Code CoAPCode = null;
 
-	//Include the context id in messages generated using this context
-	private boolean includeContextId;
+    /**
+     * Include the context id in messages generated using this context.
+     * This is generally optional and can be controlled by the application.
+     *
+     * Default value is false.
+     */
+    private boolean includeContextId;
+
+    /**
+     * Generate a new partial IV for outgoing Response messages.
+     * If this variable is false the same nonce from the original request will be used.
+     * Otherwise a new partial IV will be generated by the sender and included in the Response.
+     * This affects the calculation of the nonce.
+     *
+     * See https://tools.ietf.org/html/rfc8613#section-5.2
+     *
+     * This variable will control the behaviour when sending Response messages with this context.
+     * Note that Observe notifications will always include a new partial IV.
+     *
+     * Default value is false.
+     */
+    private boolean responsesIncludePartialIV; 
+	
+    /**
+     * URI this Context is associated with if any.
+     *
+     * That is what URI it is associated and stored under in
+     * the HashMapCtxDB.
+     */
+    private String uri;
+
+    /**
+     * String versions of the context ID, sender ID and recipient ID.
+     *
+     * These are set when the context is created (rather than for every message)
+     * to be used later when adding information about the messages to the
+     * EndpointContext on sending or receiving a message.
+     */
+    private final String contextIdString;
+    private final String senderIdString;
+    private final String recipientIdString;
```

Comment:
My editor shows
\s\s\s\s
should be
\t

Do you like to reformat it? Or should we just merge it, "as it is"?

Reply:
Fixed in commit 6b98ac703fd398530455b8c5ea3f9e617268fed0